plugin.py

```diff
@@ -757,42 +803,48 @@ def _do_karma(self, irc, channel, agent, recip, line, explicit=False):
             irc.reply("You may not modify your own karma.")
             return
 
+        release = self.get_current_release()
+
         # Check our karma db to make sure this hasn't already been done.
         data = None
         try:
             data = shelve.open(self.karma_db_path)
-            if 'forwards' not in data or 'backwards' not in data:
-                data['forwards'], data['backwards'] = {}, {}
 
-            if agent not in data['forwards']:
-                forwards = data['forwards']
+            if 'forwards-' + release not in data:
```

Comment:
when I see the number of ``'forwards_' + release`` and ``'backwards_' + release`` I wonder if we shouldn't make them variables